Apps/W1/HybridGP/app/src/Migration/Accounts/GPAccountMigrator.codeunit.al

```diff
@@ -177,4 +188,58 @@ codeunit 4017 "GP Account Migrator"
                 exit(GPGLTransactions.ACTNUMBR_8);
         end;
     end;
+
+    local procedure CreateBeginningBalance(HistYear: Integer; AccountIndex: Integer): Boolean
+    var
+        GPGL10111: Record "GP GL10111";
+        GPAccount: Record "GP Account";
+        GenJournalLine: Record "Gen. Journal Line";
+        GPFiscalPeriods: Record "GP Fiscal Periods";
+        DataMigrationFacadeHelper: Codeunit "Data Migration Facade Helper";
+        BeginningBalance: Decimal;
+        PostingGroupCode: Text;
+        FirstYear: Integer;
+    begin
+        if not GPAccount.Get(AccountIndex) then
+            exit(false);
+
+        GPGL10111.SetRange(ACTINDX, AccountIndex);
+        GPGL10111.SetRange(PERIODID, 0);
+
+        if not GPGL10111.FindSet() then
+            exit(false);
+
+        FirstYear := 0;
+        repeat
+            if GPGL10111.YEAR1 = HistYear then begin
```

Comment:
Should this section of code find the first year >= HistYear? Could this then be changed to something like this?

FirstYear := 0;
GPGL10111.SetFilter(YEAR1, '>= %1', HistYear);
GPGL10111.SetCurrentKey(YEAR1);
GPGL10111.SetAscending(YEAR1, true);
if GPGL10111.FindFirst() then begin
	FirstYear := GPGL10111.YEAR1;
	BeginningBalance := GPGL10111.PERDBLNC;
end;